### ISSUE TITLE
pouchdb-find: can't search for null

### ISSUE THREAD

**Issue by nolanlawson** [pouchdb/pouchdb#6270]
_From @froatsnook on March 27, 2016 16:26_

A query like `db.find({ selector: { day: null } })` doesn't work: (TypeError: Cannot use 'in' operator to search for '$ne' in null(…)), whereas a query like `db.find({ selector: { day: { $eq: null } } })` does!

Is this because of the Cloudant query API?  I quickly glanced at the docs and see anything.

Thanks for the library!  I started trying to do it myself (coming from Meteor), and was relieved to find this.

_Copied from original issue: nolanlawson/pouchdb-find#169_

**Comment by nolanlawson** [comment]
_From @froatsnook on March 27, 2016 17:35_

It seems there are other problems with null.

``` javascript
var db = new PouchDB("test");
db.createIndex({ index: { fields: ["field1"] } });
db.createIndex({ index: { field: ["field2"] } });
db.post({ field1: null, field2: null });
db.find({ selector: { field1: null } }); // fails as described in previous post
db.find({ selector: { field1: { $eq: null } } }).then((res) => console.log(res)); // OK, returns 1 doc
db.find({ selector: { field2: { $eq: null } } }).then((res) => console.log(res)); // OK, returns 1 doc
db.find({ selector: { field1: { $eq: null }, field2: { $eq: null } } }).then((res) => console.log(res)); // WRONG, returns empty array
```

**Comment by nolanlawson** [comment]
_From @garrensmith on March 28, 2016 12:32_

Those look like bugs. Thanks for reporting. If you could create some
failing tests in the test suite that would be really helpful.

**Comment by patrick-entinux** [comment]
Using RxDB, but it is using pouchdb-find internally as far as I understand. I had some data like:

```
{
  id: 123
}
```

And I tried to do:

```
find({createdAt: null})
```

Which got no results.

I thought this would return the above item, similar to a query for null in mongodb (https://docs.mongodb.com/manual/tutorial/query-for-null-fields/).

After that I tried:

```
find({createdAt: {$exists: false}})
```

And I got the result I hoped for. I don't know if this works in every case, but maybe good to try too.
